db-engine/src/main/java/com/splicemachine/db/catalog/types/IndexDescriptorImpl.java

```diff
@@ -269,20 +322,35 @@ else if (isUniqueWithDuplicateNulls)
 	public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
         FormatableHashtable fh = (FormatableHashtable) in.readObject();
         isUnique = fh.getBoolean("isUnique");
-        int bcpLength = fh.getInt("keyLength");
-        baseColumnPositions = new int[bcpLength];
-        isAscending = new boolean[bcpLength];
-        for (int i = 0; i < bcpLength; i++) {
-            baseColumnPositions[i] = fh.getInt("bcp" + i);
-            isAscending[i] = fh.getBoolean("isAsc" + i);
-        }
+        int keyLength = fh.getInt("keyLength");
+		isAscending = new boolean[keyLength];
+		for (int i = 0; i < keyLength; i++) {
+			isAscending[i] = fh.getBoolean("isAsc" + i);
+		}
+		int numBaseColumns = fh.containsKey("numBaseColumns") ? fh.getInt("numBaseColumns") : keyLength;
```

Comment:
we should maybe use sth like `static final String numBaseColumnsKey = "numBaseColumns"` for this (also other keys)

Reply:
Good point, would make it less error-prone. Done.